pkg/apis/workspaces/v1alpha2/volumeComponent.go

```diff
@@ -11,4 +11,9 @@ type Volume struct {
 	// +optional
 	// Size of the volume
 	Size string `json:"size,omitempty"`
+
+	// +optional
+	// Ephemeral volumes are not stored persistently across restarts. Defaults
+	// to false
+	Ephemeral bool `json:"ephemeral,omitempty"`
```

Comment:
@kadel what are your thoughts on this from odo perspective, since odo has a project preference for ephemeral volumes..

Reply:
Ephemeral preference in odo is not related to volume component in devfile. This setting is used for internal odo volume that is used for the directory where the source code is stored (`/projects`).

odo is actually treating all component volumes as not ephemeral (it always creates PVC for each defined volume component)

Reply:
odo preference may want to update the label for ephermeral as users may get confused in the future with the preference label and the devfile property on volumes.